requirements.txt

```diff
@@ -1,13 +1,15 @@
 --extra-index-url https://pypi.dwavesys.com/simple
 
-dimod==0.11.6
-dwave-preprocessing==0.4.0
-dwave-cloud-client==0.10.2
-dwave-networkx==0.8.12
+dimod==0.12.0
+dwave-preprocessing==0.5.0
+dwave-cloud-client==0.9.1
```

Comment:
Why the downgrade?

Reply:
It's the minimum version supported by the package. Figure if we're supporting it we should use it for testing etc. Happy to use a more modern version.

Reply:
Agree we should be consistent. IMO we should generally be using the minimum. But that's a bigger topic :smile:

Reply:
I think we should sweep over all latest "major" supported versions. And use latest in requirements, as they are actually dev requirements.

Reply:
But why shouldn't the dev use the oldest version supported? That way they know right away if they've accidentally relied on something newer. Also, assuming that it's likely to be the worst experience, it would make sense to me for the dev to use that experience.

Reply:
One could argue if you're developing on oldest you might not catch something is broken when latest is used (and that happens as well).

Ultimately it shouldn't really matter how dev's env looks like, as long as we sweep over `[==oldest, ~M1.0, `~M2.0`, etc]`.

Reply:
> One could argue if you're developing on oldest you might not catch something is broken when latest is used (and that happens as well).

Indeed. And it could also be that there is some middle version that's the worst experience. I think I am arguing that it's more likely that someone developing on a new version will miss something bad on an old version than the other way around. But there is no true answer other than...

> Ultimately it shouldn't really matter how dev's env looks like, as long as we sweep over [==oldest, ~M1.0, ~M2.0, etc].

:100: